uml/uml.ttl

```diff
@@ -0,0 +1,472 @@
+
+#################################################################
+#    Annotation properties
+#################################################################
+
+###  http://www.w3.org/2002/07/owl#maxCardinality
+owl:maxCardinality rdf:type owl:AnnotationProperty .
+owl:minCardinality rdf:type owl:AnnotationProperty .
+
+opil:compositionalProperty rdf:type owl:AnnotationProperty . # should be SBOL - sbol_factory issue #10
+
+
+#################################################################
+#    Datatypes
+#################################################################
+
+###  http://www.w3.org/2001/XMLSchema#anySimpleType
+xsd:anySimpleType rdf:type rdfs:Datatype .
+#sbol:Identified rdf:type owl:Class .
+#sbol:TopLevel rdf:type owl:Class .
+
+
+#################################################################
+#    Behaviors
+#################################################################
+
+# Properties for TypedElement
+uml:type rdf:type owl:DatatypeProperty ;
+         rdfs:comment "For TypedElement abstract class; UML 2.5.1 specification section 7.5" ;
+         rdfs:domain sbol:Identified ; # should probably actually be a union
+         rdfs:range xsd:anyURI ;
+         rdfs:label "type" .
+
+# Properties for MultiplicityElement
+uml:isOrdered rdf:type owl:DatatypeProperty ;
+         rdfs:comment "For MultiplicityElement abstract class; UML 2.5.1 specification section 7.5" ;
+         rdfs:domain sbol:Identified ; # should probably actually be a union
+         rdfs:range xsd:boolean ;
+         rdfs:label "is_ordered" .
+
+uml:isUnique rdf:type owl:DatatypeProperty ;
+         rdfs:comment "For MultiplicityElement abstract class; UML 2.5.1 specification section 7.5" ;
+         rdfs:domain sbol:Identified ; # should probably actually be a union
+         rdfs:range xsd:boolean ;
+         rdfs:label "is_unique" .
+
+uml:lowerValue rdf:type owl:ObjectProperty ;
+         rdfs:comment "For MultiplicityElement abstract class; UML 2.5.1 specification section 7.5" ;
+         rdfs:subPropertyOf opil:compositionalProperty ;
+         rdfs:domain sbol:Identified ; # should probably actually be a union
+         rdfs:range uml:ValueSpecification ;
+         rdfs:label "lower_value" .
+
+uml:upperValue rdf:type owl:ObjectProperty ;
+         rdfs:comment "For MultiplicityElement abstract class; UML 2.5.1 specification section 7.5" ;
+         rdfs:subPropertyOf opil:compositionalProperty ;
+         rdfs:domain sbol:Identified ; # should probably actually be a union
+         rdfs:range uml:ValueSpecification ;
+         rdfs:label "upper_value" .
+
+
+uml:ValueSpecification rdf:type owl:Class ;
+              rdfs:comment "UML 2.5.1 specification section 8" ;
+              rdfs:subClassOf sbol:Identified ,
+    # Properties inherited from TypedElement:
+    [ rdf:type owl:Restriction ; owl:allValuesFrom xsd:anyURI ; owl:onProperty uml:type ;
+      owl:maxCardinality "1"^^xsd:nonNegativeInteger ] .
```

Comment:
I believe that UML makes the type attribute optional ( 0..1 ), but do we want to? Should we just make the cardinality one and refuse to let people leave types unspecified?

Your call.

Reply:
Leaving it unspecified allows the potential for type inference to be used where possible, which I'd like to not prohibit.